projects/bms_carrier/src/main.c

```diff
@@ -1,3 +1,89 @@
+#include "bms.h"
+#include "bms_events.h"
+#include "can.h"
+#include "can_handler.h"
+#include "can_msg_defs.h"
+#include "event_queue.h"
+#include "fault_bps.h"
+#include "interrupt.h"
+#include "killswitch.h"
+#include "soft_timer.h"
+
+static CanStorage s_can_storage = { 0 };
+static BmsStorage s_bms_storage = { 0 };
```

Comment:
I **love** the architecture of having all module-level state concentrated in a single `BmsStorage` struct! Reminds me of the Redux JS framework, it could be really useful for debugging.